### PR TITLE
Dashboard: Migrate CommonJS require()s to ESM imports

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request:

Migrate CommonJS require()s to ESM imports. Produced by running `calypso-codemods commonjs-imports _inc/client/` ([see](https://github.com/Automattic/calypso-codemods#getting-started)). This seems to work surprisingly well.

This might be needed for #10810 to not break the React dashboard.

At some point in the future, we might want to re-use Calypso's (or core's) components here, but for now, I really just wanna unblock #10810 :sweat_smile: 

#### Testing instructions:

Smoke-test the heck out of Jetpack's React dashboard.

#### Proposed changelog entry for your changes:

Dashboard: Migrate CommonJS require()s to ESM imports

### CODE REVIEW

File: _inc/client/components/form/form-toggle/test/index.jsx [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-	ReactDOM = require( 'react-dom' ),
-	TestUtils = ReactDOM.TestUtils,
-	unique = require( 'lodash/uniq' );
+import assert from 'assert';
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import unique from 'lodash/uniq';
+/**
+ * External dependencies
```
Comment (sirreal) [inline _inc/client/components/form/form-toggle/test/index.jsx]: Oops

File: _inc/client/lib/analytics/index.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
 /**
  * External dependencies
  */
-const debug = require( 'debug' )( 'dops:analytics' ),
-	assign = require( 'lodash/assign' );
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'dops:analytics' );
```
Comment (sirreal) [inline _inc/client/lib/analytics/index.js]: Ideally, after imports

File: _inc/client/mixins/emitter/index.js [hunk 3]
```diff
@@ -6,8 +6,12 @@
 // load the module from `node_modules/` instead of the core’s `events.js`
 // file. Webpack uses the same module on the client side, too, which
 // makes for a nice consistency.
-const EventEmitter = require( 'events/' ).EventEmitter,
-	assign = require( 'lodash/assign' );
+
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events/';
```
Comment (sirreal) [inline _inc/client/mixins/emitter/index.js]: Strange…
```suggestion
import { EventEmitter } from 'events';
```

File: _inc/client/mixins/url-search/index.js [hunk 4]
```diff
@@ -1,13 +1,15 @@
 /**
  * External dependencies
  */
-const debug = require( 'debug' )( 'calypso:url-search' ),
-	page = require( 'page' );
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:url-search' );
```
Comment (sirreal) [inline _inc/client/mixins/url-search/index.js]: Move after imports

File: _inc/client/notices/index.js [hunk 5]
```diff
@@ -1,17 +1,18 @@
 /**
  * External Dependencies
  */
-const debug = require( 'debug' )( 'calypso:notices' );
+import debugFactory from 'debug';
 
-const Emitter = require( 'mixins/emitter' );
+const debug = debugFactory( 'calypso:notices' );
```
Comment (sirreal) [inline _inc/client/notices/index.js]: Move after imports

**Review by sirreal — COMMENTED** [review]
Inspected this visually and it all looks good. I left some nits but nothing blocking. I'll clean them up.

File: _inc/client/components/form/form-toggle/test/index.jsx [hunk 6]
```diff
@@ -1,16 +1,20 @@
...
+ */
+const TestUtils = ReactDOM.TestUtils;
```
Comment (sirreal) [inline _inc/client/components/form/form-toggle/test/index.jsx]: After imports.

File: _inc/client/components/form/form-toggle/test/index.jsx [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
 
 /**
  * Internal dependencies
  */
-const CompactFormToggle = require( 'components/forms/form-toggle/compact' );
+import CompactFormToggle from 'components/forms/form-toggle/compact';
 
 require( 'lib/react-test-env-setup' )();
```
Comment (sirreal) [inline _inc/client/components/form/form-toggle/test/index.jsx]: I can't figure out what this is referencing…

- Search indicates this is the only mention of it.
- Doesn't seem to point to any local file.

**Review by sirreal — APPROVED** [review]
I've run `yarn clean client && yarn build-client` and done a fair amount of smoke testing. Everything seems to work well.

The one place I have an issue is on the settings "Traffic" tab -> "Verify with Google" button opens a blank page that doesn't seem to load. I have the same problem on master.

**Comment by sirreal** [comment]
We still have `CommonJS` exports… it would be nice to run the codemod for those too (in the future).

**Comment by simison** [comment]
> The one place I have an issue is on the settings "Traffic" tab -> "Verify with Google" button opens a blank page that doesn't seem to load. I have the same problem on master.

@mdawaffe @eliorivero has this anything to do with Google+ deprecations?  https://github.com/Automattic/wp-calypso/issues/31299

**Review by scruffian — APPROVED** [review]
I did a smoke test and everything seems to work. The code looks good too.

**Review by brbrr — APPROVED** [review]
Works well in my tests!

File: _inc/client/components/form/form-toggle/test/index.jsx (hunk 1 above)
Comment (ockham) [reply _inc/client/components/form/form-toggle/test/index.jsx]: :blush:

File: _inc/client/components/form/form-toggle/test/index.jsx (hunk 7 above)
Comment (ockham) [reply _inc/client/components/form/form-toggle/test/index.jsx]: We had a file like that in Calypso once. If it's not in JP, probably safe to remove.

Comment (ockham) [reply _inc/client/components/form/form-toggle/test/index.jsx]: (Deferring to a follow-up PR since I don't wanna lose everyone's :heavy_check_mark:s)

**Comment by ockham** [comment]
> We still have `CommonJS` exports… it would be nice to run the codemod for those too (in the future).

Yeah, was wondering if I should include those in this PR, but then again, I thought at this point in the release cycle I should try and keep the changes to a minimum.

**Comment by ockham** [comment]
We're holding off on merging this because of the freeze.

**Review by jeherve — APPROVED** [review]
This seems to work well in my tests.

File: _inc/client/components/form/form-toggle/test/index.jsx (hunk 7 above)
Comment (ockham) [reply _inc/client/components/form/form-toggle/test/index.jsx]: #11686

**Comment by ockham** [comment]
> > We still have `CommonJS` exports… it would be nice to run the codemod for those too (in the future).
> 
> Yeah, was wondering if I should include those in this PR, but then again, I thought at this point in the release cycle I should try and keep the changes to a minimum.

#11687

**Comment by simison** [comment]
>The one place I have an issue is on the settings "Traffic" tab -> "Verify with Google" button opens a blank page that doesn't seem to load. I have the same problem on master.

Just noting that on 2nd testing, the popup works for me. It's just super slow because it redirects Ngrok -> public-api.wordpress.com -> Google.

**Pull request merged**